src/parcsr_ls/par_ilu_setup.c

```diff
@@ -3926,9 +3926,9 @@ hypre_ILUSetupMILU0(hypre_ParCSRMatrix *A, HYPRE_Int *permp, HYPRE_Int *qpermp,
          //hypre_TMemcpy(&(L_diag_j)[ctrL], iL, HYPRE_Int, lenl, HYPRE_MEMORY_DEVICE, HYPRE_MEMORY_HOST);
          //hypre_TMemcpy(&(L_diag_data)[ctrL], wL, HYPRE_Real, lenl, HYPRE_MEMORY_DEVICE, HYPRE_MEMORY_HOST);
          hypre_TMemcpy(&L_diag_j[ctrL], iL, HYPRE_Int, lenl,
-                       HYPRE_MEMORY_HOST, HYPRE_MEMORY_HOST);
+                       HYPRE_MEMORY_DEVICE, HYPRE_MEMORY_HOST);
```

Comment:
@oseikuffuor1 I restored the commented code (see above this line). Do you know why we did not want to use that?

Reply:
@oseikuffuor1 And, also the other places in this file..

Reply:
I am not sure exactly why. Must have been relying on UVM. These changes look good to me. Thanks.